library/Icinga/Web/Navigation/ConfigMenu.php

```diff
@@ -0,0 +1,293 @@
+    {
+        $count = 0;
+        $title = null;
+        $worstState = null;
+        foreach (HealthHook::collectHealthData()->select() as $result) {
+            if ($worstState === null || $result->state > $worstState) {
+                $worstState = $result->state;
+                $title = $result->message;
+                $count = 1;
+            } elseif ($worstState === $result->state) {
+                $count++;
+            }
+        }
+
+        switch ($worstState) {
+            case HealthHook::STATE_OK:
+                $count = 0;
+                break;
+            case HealthHook::STATE_WARNING:
+                $this->state = self::STATE_WARNING;
+                break;
+            case HealthHook::STATE_CRITICAL:
+                $this->state = self::STATE_CRITICAL;
+                break;
+            case HealthHook::STATE_UNKNOWN:
+                $this->state = self::STATE_UNKNOWN;
+                break;
+        }
+
+        $this->title = $title;
+
+        return $count;
+    }
+
+    protected function isSelectedItem($item)
+    {
+        if ($item !== null && Icinga::app()->getRequest()->getUrl()->matches($item['url'])) {
+            $this->selected = $item;
+            return true;
+        }
+
+        return false;
+    }
+
+    protected function createHealthBadge()
+    {
+        return $this->getHealthCount() == 0 ? '' : new StateBadge($this->getHealthCount(), 'critical');
+    }
+
+    protected function createLevel2Menu()
+    {
+        $level2Nav = HtmlElement::create(
+            'div',
+            Attributes::create(['class' => 'nav-level-2 flyout'])
+        );
+
+        $this->assembleLevel2Nav($level2Nav);
+
+        return $level2Nav;
+    }
+
+    protected function createLevel2MenuItem($item, $key)
+    {
+        if (isset($item['permission']) && ! Auth::getInstance()->hasPermission($item['permission'])) {
+            return null;
+        }
+
+        $healthBadge = null;
+        $class = null;
+        if ($key === 'health') {
+            $class = ' badge-nav-item';
+            $healthBadge = $this->createHealthBadge();
+        }
+
+        $li = HtmlElement::create(
+            'li',
+            isset($item['atts']) ? $item['atts'] : [],
+            [
+                HtmlElement::create(
+                    'a',
+                    Attributes::create(['href' => $item['url']]),
+                    [
+                        t($item['label']),
+                        isset($healthBadge) ? $healthBadge : ''
+                    ]
+                ),
+            ]
+        );
+        $li->addAttributes(['class' => 'nav-item' . $class]);
+
+        if ($this->isSelectedItem($item)) {
+            $li->addAttributes(['class' => 'selected']);
+            $this->cogItemActive = true;
+        }
+
+        return $li;
+    }
+
+    protected function createUserMenuItem()
+    {
+        $userMenuItem = HtmlElement::create('li', ['class' => 'nav-item'], []);
```

Comment:
Don't define the third argument if it's empty.

Reply:
✅